src/ripple/basics/impl/PerfLogImp.cpp

```diff
@@ -79,89 +80,82 @@ PerfLogImp::Counters::countersJson() const
     Rpc totalRpc;
     for (auto const& proc : rpc_)
     {
-        Json::Value p(Json::objectValue);
+        Rpc value;
         {
-            auto const sync =
-                [&proc]() -> boost::optional<Counters::Rpc::Sync> {
-                std::lock_guard lock(proc.second.mut);
-                if (!proc.second.sync.started && !proc.second.sync.finished &&
-                    !proc.second.sync.errored)
-                {
-                    return boost::none;
-                }
-                return proc.second.sync;
-            }();
-            if (!sync)
+            std::lock_guard lock(proc.second.mutex);
+            if (!proc.second.value.started && !proc.second.value.finished &&
+                !proc.second.value.errored)
+            {
                 continue;
-
-            p[jss::started] = std::to_string(sync->started);
-            totalRpc.sync.started += sync->started;
-            p[jss::finished] = std::to_string(sync->finished);
-            totalRpc.sync.finished += sync->finished;
-            p[jss::errored] = std::to_string(sync->errored);
-            totalRpc.sync.errored += sync->errored;
-            p[jss::duration_us] = std::to_string(sync->duration.count());
-            totalRpc.sync.duration += sync->duration;
+            }
+            value = proc.second.value;
         }
```

Comment:
What do you think about something like this?
```
        Rpc const value = [&] {
            std::lock_guard lock(proc.second.mutex);
            return proc.second.value;
        }();

        if (!value.started && !value.finished && !value.errored)
        {
            continue;
        }
```
This will sometimes save a default construction, makes the var `const`, and makes it clearer  that `value` is just a copy of `proc.second.second`. OTOH, it will sometimes copy in the vaues of `proc.second.value` when it doesn't need to. I'm fine with it as-is if you prefer. Your call.